### PR TITLE
(#134) R4R tests

### LINKED ISSUE #134: Migration of R4R tests
## As a user, I want to be able to use resources for researches app's features 

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin

Feature: Resources For Researchers app is integrated within CGOV platform

    Scenario: User is able to access a default (or “landing”) page that displays all of the expected R4R elements
        Given screen breakpoint is set to "desktop"
        And user is navigating to "/research/resources"
        Then page title is "Resources for Researchers"
        And introductory text "Resources for Researchers is a directory of NCI-supported tools and services for cancer researchers. Most resources are free of cost and available to anyone. Learn more about Resources for Researchers." appears
        And "Learn more about Resources for Researchers." text is a link with href "https://www.cancer.gov/research/resources/about"
     
Scenario: Search For Resources section
    Given screen breakpoint is set to "mobile"
        And user is navigating to "/research/resources"
        Then page title is "Resources for Researchers"
        And "Search For Resources" section is displayed with correct icon- _need to think how to test this_
        And search field is displayed with placeholder text "Search Resources for Researchers"


Scenario: Browse resources section
    Given screen breakpoint is set to "desktop"
        And user is navigating to "/research/resources"
        Then page title is "Resources for Researchers"
        And section titled "Browse Resources" is displayed
And text "Browse resources by tool type or research area or view all (/d{3}) resources." is displayed
And "view all (/d{3}) resources" text is a link with href "/research/resources/search?from=0"
        And "Browse Resources" section is displayed with correct icon - _need to think how to test this_
   
Scenario: Tool type section
    Given screen breakpoint is set to "tablet"
        And user is navigating to "/research/resources"
        Then page title is "Resources for Researchers"
And section titled "Tool Type" is displayed
        And "Tool Type" section displays the following categories
            | tools                         |
            | Analysis Tools (\d+)           |
            | Datasets & Databases (\d+)      |
            | Lab Tools (\d+)               |
            | Community Research Tools (\d+) |
            | Terminology (\d+)              |
            | Networks/Consortiums (\d+)     |
            | Clinical Research Tools (\d+)  |
            | Training Resources (\d+)      |

Scenario: Research Area section
    Given screen breakpoint is set to "mobile"
        And user is navigating to "/research/resources"
        Then page title is "Resources for Researchers"
And section titled "Research Area" is displayed
        And "Research Area" section displays the following categories
            | areas                          |
            | Cancer Biology (\d+)        |
            | Cancer Omics (\d+)             |
            | Cancer Treatment (\d+)       |
            | Screening & Detection (\d+)    |
            | Cancer Health Disparities (\d+) |
            | Causes of Cancer (\d+)         |
            | Cancer Diagnosis (\d+)         |
            | Cancer & Public Health (\d+)    |
            | Cancer Statistics (\d+)         |
            | Cancer Prevention (\d+)         |
            | Bioinformatics (\d+)          |
            | Cancer Survivorship (\d+)       |
    Then the page options should appear at the bottom of the page
        And the following page options are displayed
            | pageOption |
            | Twitter    |
            | Email      |


    Scenario: User is able to see all the resources on View all Resources page
        Given screen breakpoint is set to "desktop"
        Given user is navigating to "/research/resources"
        When user clicks on "view all (/d+) resources." link
Then user is redirected to "research/resources/search?from=0"
        And page title is "Resources for Researchers Search Results"
And "Resources for Researchers Home" link has a href "/research/resources/"
        And search box is empty
        And results info text is "1 - 20 of /d+ results"
        And both pagers are displayed
        And the results are displayed with each title containing a link
       And each result has a description text under the title


    Scenario: User is able to search resources using the tool type options on the homepage
        Given screen breakpoint is set to "desktop"
        Given user is navigating to "/research/resources"
        When user clicks on "<tools>"
        Then user is redirected to "<url>"
        And the filter displays the "<toolTypeFilter>"
        And search box is empty
        And results info text is "1 - 20 of /d+ results"
        And the tool type box displays the filter "<tools>"
        Examples:

            | tools                         | toolTypeFilter           | url                                               |
            | Analysis Tools (/d+)          | Analysis Tools           | /search?from=0&toolTypes=analysis_tools           |
            | Datasets & Databases (/d+)     | Datasets & Databases     | /search?from=0&toolTypes=datasets_databases       |
            | Lab Tools (/d+)                | Lab Tools                | /search?from=0&toolTypes=lab_tools                |
            | Community Research Tools (/d+) | Community Research Tools | /search?from=0&toolTypes=community_research_tools |
            | Terminology (/d+)              | Terminology              | /search?from=0&toolTypes=terminology              |
            | Networks/Consortiums (/d+)     | Networks/Consortiums     | /search?from=0&toolTypes=networks_consortiums     |
            | Clinical Research Tools (/d+)  | Clinical Research Tools  | /search?from=0&toolTypes=clinical_research_tools  |
            | Training Resources (/d+)        | Training Resources       | /search?from=0&toolTypes=training_resources       |

    Scenario: User is able to search resources using the research area options on the homepage
        Given screen breakpoint is set to "desktop"
        Given user is navigating to "/research/resources"
        When user clicks on "<areas>"
        Then user is redirected to "<url>"
        And the filter displays the "<researchAreaFilter>"
        And search box is empty
        And Result Count text contains the same number as present in the "<areas>" link
        And the tool type box displays the filter "<areas>"
        Examples:
            | areas                          | researchAreaFilter        | url                                                    |
            | Cancer Biology (/d+)           | Cancer Biology            | /search?from=0&researchAreas=cancer_biology            |
            | Cancer Omics (/d+)              | Cancer Omics              | /search?from=0&researchAreas=cancer_omics              |
            | Cancer Treatment (/d+)          | Cancer Treatment          | /search?from=0&researchAreas=cancer_treatment          |
            | Screening & Detection (/d+)     | Screening & Detection     | /search?from=0&researchAreas=screening_detection       |
            | Cancer Health Disparities (/d+) | Cancer Health Disparities | /search?from=0&researchAreas=cancer_health_disparities |
            | Causes of Cancer (/d+)          | Causes of Cancer          | /search?from=0&researchAreas=causes_of_cancer          |
            | Cancer Diagnosis (/d+)          | Cancer Diagnosis          | /search?from=0&researchAreas=cancer_diagnosis          |
            | Cancer & Public Health (/d+)    | Cancer & Public Health    | /search?from=0&researchAreas=cancer_public_health      |
            | Cancer Statistics (/d+)         | Cancer Statistics         | /search?from=0&researchAreas=cancer_statistics         |
            | Cancer Prevention (/d+)         | Cancer Prevention         | /search?from=0&researchAreas=cancer_prevention         |
            | Bioinformatics (/d+)            | Bioinformatics            | /search?from=0&researchAreas=bioinformatics            |
            | Cancer Survivorship (/d+)       | Cancer Survivorship       | /search?from=0&researchAreas=cancer_survivorship       |

```
```gherkin
  Scenario: User is able to search resources using the search box and clicking on magnifying glass icon on the homepage
        Given screen breakpoint is set to "mobile"
        Given user is navigating to "/research/resources"
        When user types "asarp" in the search bar
        When user clicks on the magnifying glass icon
        Then user is redirected to "/search?from=0&q=asarp"
        And search box is displays the term "asarp"
        And the tooltype box displays the filter "Analysis Tools (1)"
        And the Research Area box is displayed
        And the Research Type box is displayed
        And the Tool Subtype box is displayed
        And the search result is displayed
        And the search result title is "Allele-Specific Alternative mRNA processing (ASARP)" with href "/research/resources/resource/102"
        And the search result description is displayed

    Scenario: User is able to search resources using the search box and hitting the Enter key on the homepage
        Given screen breakpoint is set to "desktop"
        Given user is navigating to "/research/resources"
        When user types "asarp" in the search bar
        When user hits "enter" key
        Then user is redirected to "/search?from=0&q=asarp"
        And search box is displays the term "asarp"
        And the tooltype box displays the filter "Analysis Tools (1)"
        And the Research Area box is displayed
        And the Research Type box is displayed
        And the Tool Subtype box is displayed
        And the search result is displayed
        And the search result title is "Allele-Specific Alternative mRNA processing (ASARP)" with href "/research/resources/resource/102"
        And the search result description is displayed
        And "top" pager is displayed
        And "bottom" pager is displayed

    Scenario: User is able to see the No results found page
        Given screen breakpoint is set to "desktop"
        Given user is navigating to "/research/resources"
        When user types "fre" in the search bar
        When user hits "enter" key
        Then user is redirected to "/search?from=0&q=fre"
        And page title is "Resources for Researchers Search Results"
        And search box displays the term "fre"
        And the message is displayed as "No results were found for your search. Please try another search or view all resources."
        And "view all (/d{3}) resources." link has a href "/research/resources/search?from=0"
        And the following page options are displayed
            | pageOption |
            | Twitter    |
            | Email      |

    Scenario Outline: User is able to see the Resource Detail page
        Given screen breakpoint is set to "desktop"
        Given user is navigating to "<url>"
        Then the resource title s displayed as "<title>"
        And the link "Resources for Researchers Home" is displayed with href "/research/resources"
        And the resource description is displayed
        And "Visit Resource" button is displayed
        And search box is displayed
        And "Find Related Resources" section is displayed
        And "Resource Access" note and indicator is displayed
        And "NCI Affliliation" field is displayed
        And "Contact Information" field is displayed

        Examples:
            | url                              | title                                                 |
            | /research/resources/resource/82  | Health Disparities Calculator (HD*Calc)               |
            | /research/resources/resource/102 | Allele-Specific Alternative mRNA processing (ASARP)   | 
            | /research/resources/resource/32  | The International Registry of Werner Syndrome         |
            | /research/resources/resource/54  | Cancer Intervention and Surveillance Modeling Network |


    Scenario: User is able to go to Resource Detail page from Search Results page
        Given screen breakpoint is set to "desktop"
        Given user is navigating to "/research/resources"
        When user clicks on "Cancer Biology"
        Then user is redirected to "/search?from=0&researchAreas=cancer_biology"
        And the filter displays the "Cancer Biology"
        And search box is empty
        And results info text is "1 - 20 of /d+ results"
        And the research area box displays the filter "Cancer Biology" as checked
        When user clicks on 1 search result item
        Then the page title is the same as the clicked result's item title
        And the "Back to Search Results" is displayed
        And the link "Resources for Researchers Home" is displayed with href "/research/resources"
        And the resource description is displayed
        And "Visit Resource" button is displayed with exit disclaimer icon
        And search box is displayed
        And "Find Related Resources" section is displayed
        And "Resource Access" note and indicator is displayed
        And "NCI Affiliation" field is displayed
        And "Contact Information" field is displayed

    Scenario: User is able to see the Resource Not Found page when non existent page is searched
        Given screen breakpoint is set to "desktop"
        When user is navigating to "/research/resources/search?from=999"
        Then the page title is "Resource Not Found" 
        And the message is displayed as "We can't find the resource you're looking for. Please visit the Resources for Researchers home page to search or browse resources."
        And "Resources for Researchers home page" link has a href "/research/resources/"
        And the following page options are displayed
            | pageOption |
            | Twitter    |
            | Email      |
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by SupriyaSaxena0605**
Added feature file.

Closes #134.

### CODE REVIEW

File: cypress/integration/ResourcesForResearchers/index.js [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+});
+
+And('results info text is {string}', (resultsInfo) => {
+  const regex = new RegExp(resultsInfo);
+  cy.get('[aria-label="Results count"] p').invoke('text').should('match', regex);
+});
+
+And('both pagers are displayed', () => {
+  cy.get('.r4r-results.r4r-DEFAULT>nav').should('have.length', 2);
+  cy.get('.r4r-results.r4r-DEFAULT>nav').each(($el) => {
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:99]: duplicating locators ! Use alias instead https://docs.cypress.io/api/commands/as

or second cy.get can be actually chained off the first like so 

```
 cy.get(".r4r-results.r4r-DEFAULT>nav")
    .should("have.length", 2)
    .each(($el) => {
      expect($el).to.be.visible;
    });
```

File: cypress/integration/ResourcesForResearchers/index.js [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    expect('div r4r-pager__count  r4r-DEFAULT').to.exist;
+    expect('div r4r-pager__nav  r4r-DEFAULT').to.exist;
+  });
+
+});
+
+And('the results are displayed with each title containing a link', () => {
+  cy.get('[aria-label="search results"]').should('be.visible');
+  cy.get('[aria-label="search results"] > article>h2').first().should('be.visible').find('a').should('have.attr', 'href');
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:108]: the AC clearly says : ... **each** title containing a link - why are you only verifying the first one?

Comment (kate-mashkina) [reply cypress/integration/ResourcesForResearchers/index.js:108]: duplicated locators again, please use alias

Comment (kate-mashkina) [reply cypress/integration/ResourcesForResearchers/index.js:108]:
```
 cy.get('section[aria-label="search results"]')
    .as("searchresults")
    .should("be.visible");
  cy.get("@searchresults")
    .find("article a")
    .each(($el) => {
      cy.wrap($el).invoke("text").should("not.be.empty");
      cy.wrap($el).should("have.attr", "href");
    });
```

File: cypress/integration/ResourcesForResearchers/index.js [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+});
+
+And('each result has a description text under the title', () => {
+  cy.get('[aria-label="search results"] > article>p').first().should('be.visible');
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:112]: same as above - AC asks to check every single result, not only the first one...

```
  cy.get('section[aria-label="search results"] > article>p').each(($el) => {
    expect($el[0].textContent).to.not.be.empty;
  });
```

File: cypress/integration/ResourcesForResearchers/index.js [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+});
+
+And('the filter displays the {string}', (filter) => {
+  cy.get('[aria-label="Selected Search Filters"] [tabindex] p').should('contain.text', filter);
+});
+
+And('the {string} box displays the filter {string}', (section, tool) => {
+  const regex = new RegExp(tool);
+  cy.get('.facet__box--tool-types').should('be.visible').find('h4').should('contain.text', section);
+  cy.get('.r4r-DEFAULT.facet__box.facet__box--tool-types > label[role="checkbox"] > .r4r-DEFAULT.filter__label').invoke('text').should('match', regex);
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:126]: this locator is waaay too long and it's first part points to the first cy.get locator , so alias again)

```
cy.get('div[class*="box--tool-types"]')
    .as("toolType")
    .find("h4")
    .should("be.visible")
    .and("contain.text", section);
  cy.get("@toolType").find("span").invoke("text").should("match", regex);
```

File: cypress/integration/ResourcesForResearchers/index.js [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+
+});
+
+And('the {string} box is displayed for {string}', (section, eligibleToolType) => {
+  if (eligibleToolType == "Y") {
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:131]: JavaScript equal operator is actually a `===`

Comment (kate-mashkina) [reply cypress/integration/ResourcesForResearchers/index.js:131]: this should make use of `section` parameter passed in ( no unused parameters)

```
And("the {string} box is displayed for {string}", (section, toolTypes) => {
  if (toolTypes === "Y") {
    cy.get("h4").contains(section).should("be.visible");
  } else {
    cy.get("h4").contains(section).should("not.exist");
  }
});
```

File: cypress/integration/ResourcesForResearchers/index.js [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+    cy.get('.facet__box--tool-subtypes').should('be.visible');
+  }
+});
+
+And('the {string} box displays the correct {string} checked', (section, area) => {
+  const regex = new RegExp(area);
+  cy.get('.r4r-DEFAULT.facet__box> h4').eq[2].should('contain', section);;
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:138]: this step does not exist in the feature file - remove it

File: cypress/integration/ResourcesForResearchers/index.js [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+  cy.get('.searchbar__container button').click();
+});
+
+And("search box displays the term {string}", (searchText) => {
+  cy.get('.searchbar__container input').should('have.attr', 'value', searchText);
+});
+
+When('user hits {string} key', (enterKey) => {
+  cy.get('.searchbar__container input').type('{enter}{enter}');
+  cy.wait(2000);
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:161]: better to remove hardcoded wait here and add a step in the feature file after this step
`And browser waits`

File: cypress/integration/ResourcesForResearchers/index.js [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+Then('the resource title is displayed as {string}', (resourceTitle) => {
+  cy.get('.r4r-resource__header h1').should('contain.text', resourceTitle);
+});
+
+And("{string} is displayed with href {string}", (linkText, linkHref) => {
+  cy.get('.r4r-DEFAULT.resource__home > a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+// And('the {string} is displayed', () => {
+//   cy.get('[aria-label="Resource description"]>p').should('be.visible');
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:197]: remove commented out step

File: cypress/integration/ResourcesForResearchers/index.js [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+And('Visit Resource link is displayed', () => {
+  cy.get('.resource__link--external').should('be.visible');
+});
+
+And('search box is displayed', () => {
+  cy.get('.searchbar__container input').should('be.visible');
+});
+
+And('{string} section is displayed', (section) => {
+  cy.get('.resource__nav h2').should('be.visible');
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:209]: no unused variables - section parameter should be incorporated into the code
cy.get('.resource__nav h2').contains(section).should('be.visible');

File: cypress/integration/ResourcesForResearchers/index.js [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+});
+
+And('{string} field is displayed', (fieldName) => {
+  cy.document().then(doc => {
+    expect(doc.querySelector(`[aria-label="${fieldName}"]`)).to.exist;
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:214]: no need to use cy.document here

` cy.get(`article[aria-label="${fieldName}"]`).should('be.visible');`

File: cypress/integration/ResourcesForResearchers/index.js [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+  })
+});
+
+And('Contact Information field is displayed', (fieldName) => {
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:218]: this becomes 

And("Contact Information field is {string}", (visibility) => {
  if(visibility.toLowerCase() === 'visible')
  cy.get('h2').contains('Contact Information').should('be.visible');
  else 
  cy.get('h2').contains('Contact Information').should('not.exist');
  
});

File: cypress/integration/ResourcesForResearchers/index.js [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+  })
+});
+
+And("the {string} is displayed with href {string}", (linkText, linkHref) => {
+  cy.get('.r4r-container.row a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+And('the {string} box displays the filter {string} as checked', (section, area) => {
+  cy.document().then(doc => {
+    const box = doc.querySelectorAll(`.r4r-DEFAULT.facet__box >h4`);
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:242]: no need to use cy.document

10 lines of code becomes 1 : 
`cy.get('h4').contains(section).parent().find(`input[value="${area}"]`).should('be.checked')`

File: cypress/integration/ResourcesForResearchers/index.js [hunk 13]
```diff
@@ -0,0 +1,264 @@
...
+      expect(box[1]).to.contain(section);
+      expect(doc.querySelector(`.r4r-DEFAULT.facet__box >label[aria-checked="true"]>input`)).to.have.property('value', area);
+    } else if (section == "Research Type") {
+      expect(box[2]).to.contain(section);
+      expect(doc.querySelector(`.r4r-DEFAULT.facet__box >label[aria-checked="true"] >input`)).to.have.property('value', area);
+    }
+  })
+});
+
+When('user clicks on 1 search result item', () => {
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:253]: number should not be hardcoded, it is a dynamic parameter 

```
When("user clicks on {int} search result item", (resIndex) => {
  cy.get(".r4r-DEFAULT.results-container >article> h2 > a").eq(resIndex-1).click();
});
```

File: cypress/integration/ResourcesForResearchers/index.js [hunk 14]
```diff
@@ -0,0 +1,264 @@
...
+  cy.get('.r4r-DEFAULT.results-container >article> h2 > a').first().click();
+});
+
+Then('the {string} is displayed', () => {
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers/index.js:257]: you are passing a string parameter , but not? where did it go?

Comment (kate-mashkina) [reply cypress/integration/ResourcesForResearchers/index.js:257]:
```
Then("the {string} is displayed", (backToSearch) => {
  cy.get(".r4r-DEFAULT.resource__back").contains(backToSearch).should("be.visible");
});
```

File: cypress/integration/ResourcesForResearchers.feature [hunk 15]
```diff
@@ -0,0 +1,229 @@
...
+        Given user is navigating to "/research/resources"
+        When user clicks on "Cancer Biology"
+        Then user is redirected to "/research/resources/search" with "?from=0&researchAreas=cancer_biology"
+        And the filter displays the "Cancer Biology"
+        And search box is empty
+        And results info text is "1 - 20 of \d+ results"
+        And the "Research Area" box displays the filter "Cancer Biology" as checked
+        When user clicks on 1 search result item
+        # Then the page title is the same as the clicked result's item title
+        And the "Back to Search Results" is displayed
```
Comment (kate-mashkina) [inline cypress/integration/ResourcesForResearchers.feature:210]: this should be 
 And the "< Back to Results" is displayed

**Review by kate-mashkina — CHANGES_REQUESTED** [review]

File: cypress/integration/ResourcesForResearchers/index.js (hunk 14 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers/index.js:257]: fixed it

File: cypress/integration/ResourcesForResearchers/index.js (hunk 13 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers/index.js:253]: fixed it

File: cypress/integration/ResourcesForResearchers/index.js (hunk 10 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers/index.js:214]: thanks for your help.

File: cypress/integration/ResourcesForResearchers/index.js (hunk 11 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers/index.js:218]: fixed

File: cypress/integration/ResourcesForResearchers/index.js (hunk 12 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers/index.js:242]: fixed it, thanks

File: cypress/integration/ResourcesForResearchers/index.js (hunk 7 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers/index.js:161]: done

File: cypress/integration/ResourcesForResearchers/index.js (hunk 5 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers/index.js:131]: fixed it

File: cypress/integration/ResourcesForResearchers/index.js [hunk 16]
```diff
@@ -0,0 +1,264 @@
...
+  cy.location('pathname').should('eq', redirectedPath);
+  cy.location('search').should('include', urlSearch);
+});
+
+And("{string} link has a href {string}", (linkText, linkHref) => {
+  cy.get('header > a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+And("search box is empty", () => {
+  cy.get('.searchbar__container input').should('contain.text', "");
```
[the remark this answers is not in the document]
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers/index.js:89]: fixed it

File: cypress/integration/ResourcesForResearchers/index.js [hunk 17]
```diff
@@ -0,0 +1,264 @@
...
+
+And('the following {string} are displayed', (pageOption) => {
+  const options = document.querySelectorAll(`div#PageOptionsControl1 > ul`);
+  for (let i = 0; i < options; i++) {
+    cy.get('div#PageOptionsControl1 > ul>li>a').should('contain.text', title);
+  }
+});
+
+And("user clicks on {string} link", (viewAll) => {
+  cy.get('.r4r-DEFAULT.view-all__link').click();
```
[the remark this answers is not in the document]
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers/index.js:76]: fixed it

File: cypress/integration/ResourcesForResearchers/index.js [hunk 18]
```diff
@@ -0,0 +1,264 @@
...
+
+});
+
+And('{string} is a link', (viewAll) => {
+  const regex = new RegExp(viewAll);
+  cy.get('.r4r-DEFAULT.view-all__link').invoke('text').should('match', regex);
+
+});
+
+And('{string} section displays the following {string} categories', (section, options) => {
```
[the remark this answers is not in the document]
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers/index.js:47]: fixed it

File: cypress/integration/ResourcesForResearchers/index.js [hunk 19]
```diff
@@ -0,0 +1,264 @@
...
+
+  cy.get('.home-tile__text').contains(title).parent().as('titleText').should('be.visible');
+  cy.get('@titleText').parent().find('.home-tile__icon svg').should('be.visible');
+});
+
+And('search field is displayed with placeholder text {string}', (placeholder) => {
+  cy.get('.searchbar__container input').should('have.attr', 'placeholder', placeholder);
+});
+
+And('section titled {string} is displayed', (title) => {
```
[the remark this answers is not in the document]
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers/index.js:23]: fixed it

File: cypress/integration/ResourcesForResearchers.feature [hunk 20]
```diff
@@ -0,0 +1,229 @@
...
+        And "Resources for Researchers Home" is displayed with href "/research/resources/"
+        And "Resource description" field is displayed
+        And Visit Resource link is displayed with exit disclaimer icon
+        And search box is displayed
+        And "Find Related Resources" section is displayed
+        And "Resource Access Information" field is displayed
+        And "NCI Affiliation Information" field is displayed
+        And Contact Information field is displayed
```
[the remark this answers is not in the document]
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers.feature:218]: done

File: cypress/integration/ResourcesForResearchers.feature (hunk 15 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers.feature:210]: done

File: cypress/integration/ResourcesForResearchers.feature [hunk 21]
```diff
@@ -0,0 +1,229 @@
...
+        Given user is navigating to "<url>"
+        Then the resource title is displayed as "<title>"
+        And "Resources for Researchers Home" is displayed with href "/research/resources/"
+        And "Resource description" field is displayed
+        And Visit Resource link is displayed
+        And search box is displayed
+        And "Find Related Resources" section is displayed
+        And "Resource Access Information" field is displayed
+        And "NCI Affiliation Information" field is displayed
+        And Contact Information field is displayed
```
[the remark this answers is not in the document]
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers.feature:189]: done

File: cypress/integration/ResourcesForResearchers.feature [hunk 22]
```diff
@@ -0,0 +1,229 @@
...
+    Scenario Outline: User is able to search resources using the tool type options on the homepage
+        Given screen breakpoint is set to "desktop"
+        Given user is navigating to "/research/resources"
+        When user clicks on "<toolTypeFilter>"
+        Then user is redirected to "<redirectedPath>" with "<search>"
+        And the filter displays the "<toolTypeFilter>"
+        And search box is empty
+        And results info text is "1 - \d+ of \d+ results"
+        And the "Tool Type" box displays the filter "<tools>"
+        And the "Tool Sub-Type" box is displayed for "<eligibleTooltypes>"
```
[the remark this answers is not in the document]
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers.feature:93]: fixed it

File: cypress/integration/ResourcesForResearchers.feature [hunk 23]
```diff
@@ -0,0 +1,229 @@
...
+        And the following "<pageOption>" are displayed
+            | pageOption |
+            | Twitter    |
+            | Email      |
+
+
+    Scenario: User is able to see all the resources on View all Resources page
+        Given screen breakpoint is set to "desktop"
+        Given user is navigating to "/research/resources"
+        When user clicks on "view all (\d+) resources." link
```
[the remark this answers is not in the document]
Comment (SupriyaSaxena0605) [reply cypress/integration/ResourcesForResearchers.feature:73]: done

**Comment by SupriyaSaxena0605** [comment]
@KateMashkinaNIH Please review. I have made all the changes.
image

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**